modules/Bio/EnsEMBL/Hive/DBSQL/AnalysisJobAdaptor.pm

```diff
@@ -587,6 +587,54 @@ sub reset_or_grab_job_by_dbID {
     return $job;
 }
 
+=head2 reset_job_by_input_id_and_sync
+
+  Arg [1]: string $input_id pattern
+  Arg [2]: array $analysis_id
+  Example:
+    my $job = $self->{'dba'}->get_AnalysisJobAdaptor->reset_job_by_input_id_and_sync($input_id_pattern, $analyses_pattern);
+  Description:
+    Reset jobs for the specified $input_id pattern and analyses_id pattern. $input_id can be a wildcard argument.
+  Returntype : none
+  Exceptions :
+  Caller     : beekeeper.pl
+
+=cut
+
+sub reset_job_by_input_id_and_sync {
+    my ($self, $input_id_pattern, @analyses_list) = @_;
+    $input_id_pattern =~ s/\%/\.*/g;
+    my $analysis;
+    my $jobs = $self->fetch_all_by_analysis_id_status(@analyses_list);
+    foreach my $job (@$jobs){
+         my $input_id = $job->input_id;
+         if ($input_id =~ /$input_id_pattern/) {
+                my $job_status = $job->status();
+
+                if($job_status =~ $ALL_STATUSES_OF_RUNNING_JOBS) {
+                       die "Job is already in progress, cannot reset";
+                }
+
+                if(($job_status eq 'DONE') and my $controlled_semaphore = $job->controlled_semaphore) {
+                       $controlled_semaphore->increase_by( [ $job ] );
+                }
+                $analysis = $job->analysis;
+                $job->set_and_update_status('READY');
+                $analysis->stats->adaptor->increment_a_counter( $Bio::EnsEMBL::Hive::AnalysisStats::status2counter{$job->status}, 1, $job->analysis_id );
+
+                # Syncing stats. Copied from synchronize_AnalysisStats()
+                my $stats = $job->analysis->stats;
+                $stats->refresh();
+
+                my $job_counts = $stats->hive_pipeline->hive_use_triggers() ? undef : $self->fetch_job_counts_hashed_by_status( $stats->analysis_id );
+
+                $stats->recalculate_from_job_counts( $job_counts );
+
+                $stats->update;
```

Comment:
I think this section (since setting `$stats`) is not needed since you already take care of the counters above (once you decrement the other one)